### ISSUE TITLE
[BUG] const uint8_t[:, :] as shape [1, 2, 0, 0, 0, 0, 0, 0]

### ISSUE THREAD

**Issue by jbrockmendel** [cython/cython#4321]
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=63805&view=logs&j=25ded215-59e0-5b6f-7d33-3c0cd0788e22&t=a63eee02-97df-5ea7-0cd1-dfa775102e83&l=70

I have a function where the `mask` argument is declared as `const uint8_t[:, :]` and a validation that mask.shape[1] should only ever be 1, but I'm getting mask.shape or [1, 2, 0, 0, 0, 0, 0, 0] in some builds.  AFAICT this is only in cases where mask _should_ be None.

**Comment by 0dminnimda** [comment]
It would be nice if you add a minimal working example.

**Comment by da-woods** [comment]
The trailing zeros are just a consequence of memoryviews having a fixed number of dimensions so they're expected.

I dunno about the `shape[1]` validation. Possibly the `shape` isn't initialized when the memoryview is set to `None`?

**Comment by jbrockmendel** [comment]
> It would be nice if you add a minimal working example.

agreed!  ATM this is only happening on ~half of CI builds and I can't reproduce it locally.  If that changes I'll update.

**Comment by scoder** [comment]
I agree that this is annoying. We should fix it. See the examples in
https://groups.google.com/d/msgid/cython-users/a7f3bd42-8530-464c-8d86-246dbbe17c57%40googlegroups.com

**Comment by scoder** [comment]
Also see #863.
